plugins/rabbitmq/src/main/java/com/navercorp/pinpoint/plugin/rabbitmq/client/interceptor/ChannelBasicPublishInterceptor.java

```diff
@@ -162,16 +162,16 @@ private boolean validate(Object target, Object[] args) {
         if (args == null || args.length < 6) {
             return false;
         }
-        if (!(args[0] instanceof String)) {
+        if (args[0] !=null && !(args[0] instanceof String)) {
             return false;
         }
-        if (!(args[1] instanceof String)) {
+        if (args[1] !=null && !(args[1] instanceof String)) {
```

Comment:
Allowing **null** routingKey will open up a window for NPE in the `after()` method above :
```
String routingKey = (String) args[1];
...
    ...
    recorder.recordAttribute(RabbitMQClientConstants.RABBITMQ_ROUTINGKEY_ANNOTATION_KEY, routingKey);
```

I think you should decide whether to skip or trace when routingKey is **null**.
If you'd like to skip, then I think validation should fail when `args[1] == null`.
If you decide to trace, then routingKey should be converted to a non-null (an empty string?) above.

Reply:
args[0] is exhange and args[1] is routingKey, these two args should not be null which will cause exception by amqp-client 3.5.5. So it would not cause NPE in after(). So we need not check them against null.